Library/Homebrew/cmd/install.rb

```diff
@@ -317,11 +397,11 @@ def install_formula(f)
     fi = FormulaInstaller.new(f)
     fi.options              = build_options.used_options
     fi.invalid_option_names = build_options.invalid_option_names
```

Comment:
Might be a good idea to use the `build_options.invalid_option_names` above instead of the formulae loop?

Reply:
Sorry, I'm not sure I understand. Can you explain this more?

Reply:
@GauthamGoli If we're only looking at externally passed options I think it's fine/better to error out rather than ignoring and messaging 👍